core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -362,26 +352,6 @@ Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
      */
     OptionalLong executeDelete(Session session, TableHandle tableHandle);
 
-    /**
-     * Begin delete query
-     */
-    TableHandle beginDelete(Session session, TableHandle tableHandle);
-
-    /**
-     * Finish delete query
-     */
-    void finishDelete(Session session, TableHandle tableHandle, Collection<Slice> fragments);
-
-    /**
-     * Begin update query
-     */
-    TableHandle beginUpdate(Session session, TableHandle tableHandle, List<ColumnHandle> updatedColumns);
```

Comment:
The `beginMerge` method doesn't have a list of updated columns, so `beginMerge` can't replace it in all cases. I opened #15585 with more details about my use case so it's easier to discuss it here, but IMO we should resolve it before merging this PR.